### PR TITLE
Add deployment procedure for openstack

### PR DESCRIPTION
**Pull request by r0ckarong**
Add more content to notes file

Add to notes files

Added placeholder section for openstack

Add preliminary screenshots for SOC Horizon

Add procedure for openstack

Split instructions into UI and CLI

Added OpenStack entity

Add more details to procedures

Rework sections

Rework wording for openstack sections

Rework names across examples, update screenshots

Add clean screenshots

Remove notes file

### CODE REVIEW

**Comment by r0ckarong** [comment]
Screenshots again aren't right, will rework

**Comment by r0ckarong** [comment]
Redone screenshots with Opensuse on VM

File: xml/deployment_installation.xml [hunk 1]
```diff
@@ -825,9 +825,253 @@
       configured separately and differently from the &worker_node;s.
      </para>
     </sect2>
+
+    <sect2 xml:id="sec.deploy.install.qcow2.procedure.openstack">
+     <title>Installation on &soc;</title>
+     <para>
+      You can easily deploy a &productname; on &soc; using &ostack;.
```
Comment (cjschroder) [inline xml/deployment_installation.xml]: Not sure I would say "easily" :)

File: xml/deployment_installation.xml [hunk 2]
```diff
@@ -825,9 +825,253 @@
...
+      <title>Using the Horizon Dashboard</title>
+      <procedure>
+
+      <step>
+       <para>
+        Go to <guimenu>Project -> Compute -> Images</guimenu> and click on 
+        <guimenu>Create Image</guimenu>.
+       </para>
+       <para>
+        Give your image a name (for example: <literal>CaaSP-3</literal>) you will
```
Comment (cjschroder) [inline xml/deployment_installation.xml]: Needs a conjunction, e.g. "as you will need this later"

File: xml/deployment_installation.xml [hunk 3]
```diff
@@ -825,9 +825,253 @@
...
+        <para>
+         Click on <guimenu>Launch Stack</guimenu>.
+        </para>
+       </step>
+       
+       <step>
+        <para>
+         You need to provide the stack templates at this stage.
+         Either upload the files, provide the URL to the raw files directly 
+         (only applies to stack template) or copy and paste the contents into the
```
Comment (cjschroder) [inline xml/deployment_installation.xml]: need a comma, "template), or copy and paste"

File: xml/deployment_installation.xml [hunk 4]
```diff
@@ -825,9 +825,253 @@
...
+       <para>
+        Click <guimenu>Next</guimenu>.
+       </para>
+      </step>
+
+ <!-- FIXME mnapp, 17/6/18 This screenshot has a keepassxc icon in the root password box, replace -->
+       <step>
+        <para>
+         Now you need to define more information about the stack.
+        </para>
```
Comment (cjschroder) [inline xml/deployment_installation.xml]: Maybe use `<varname/> tags here, @cwickert @lproven what say you?

Comment (cjschroder) [reply xml/deployment_installation.xml]: sigh, varname....

Comment (cjschroder) [reply xml/deployment_installation.xml]: Forgot to mention that it builds without errors.

File: xml/deployment_installation.xml [hunk 5]
```diff
@@ -825,9 +825,253 @@
...
+        <itemizedlist>
+         <listitem><para>Stack Name - Give your stack a name</para></listitem>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: listitem and para are block elements, not inline elements. So you should insert line breaks and indent the text accordingly.

File: xml/deployment_installation.xml [hunk 6]
```diff
@@ -825,9 +825,253 @@
...
+        Click <guimenu>Next</guimenu>.
+       </para>
+      </step>
+
+ <!-- FIXME mnapp, 17/6/18 This screenshot has a keepassxc icon in the root password box, replace -->
+       <step>
+        <para>
+         Now you need to define more information about the stack.
+        </para>
+        <itemizedlist>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: For lists that explain GUI items, we usually we use variablelist. Not sure how it looks if the explanations are for the items are that short.

File: xml/deployment_installation.xml [hunk 7]
```diff
@@ -825,9 +825,253 @@
...
+       
+       <step>
+        <para>
+         You need to provide the stack templates at this stage.
+         Either upload the files, provide the URL to the raw files directly 
+         (only applies to stack template) or copy and paste the contents into the 
+         <literal>Direct Input</literal> fields.
+        </para>
+        <warning>
+         <title>Replace the default <replaceable>root_password</replaceable></title>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: this would be guimenu (and optionally replacable).

File: xml/deployment_installation.xml [hunk 8]
```diff
@@ -825,9 +825,253 @@
...
+     <title>Installation on &soc;</title>
+     <para>
+      You can easily deploy a &productname; on &soc; using &ostack;. 
+      You will need a &productname; machine image and &ostack; Heat templates.
+      Once you have created a stack you will continue with the &productname; setup.
+     </para>
+     <note>
+      <title>&productname; machine image for &soc;</title>
+      <para>
+       Download the latest SUSE CaaS Platform for OpenStack image from
```
Comment (cwickert) [inline xml/deployment_installation.xml]: use the &productname; and &ostack entities.

File: xml/deployment_installation.xml [hunk 9]
```diff
@@ -825,9 +825,253 @@
...
+       </para>
+     </note>
+     
+     <sect3>
+      <title>Using the Horizon Dashboard</title>
+      <procedure>
+
+      <step>
+       <para>
+        Go to <guimenu>Project -> Compute -> Images</guimenu> and click on
```
Comment (cwickert) [inline xml/deployment_installation.xml]: Use proper &rarr; arrows with `&rarr;`

File: xml/deployment_installation.xml [hunk 10]
```diff
@@ -825,9 +825,253 @@
...
+        <guimenu>Create Image</guimenu>.
+       </para>
+       <para>
+        Give your image a name (for example: <literal>CaaSP-3</literal>) you will
+        need to use this later to find the image.
+       </para>
+      </step>
+       <step>
+        <para>
+         Go to <guimenu>Project -> Orchestration</guimenu> and click on
```
Comment (cwickert) [inline xml/deployment_installation.xml]: use proper arrows...

File: xml/deployment_installation.xml [hunk 11]
```diff
@@ -825,9 +825,253 @@
...
+        </para>
+        <warning>
+         <title>Replace the default <replaceable>root_password</replaceable></title>
+          <para>
+           Do not use the <filename>caasp-environment.yaml</filename> via it's URL
+           directly from the GitHub repository.
+          </para>
+           <para>
+            You must make sure to replace the value for 
+            <replaceable>root_password</replaceable> with a secure password.
```
Comment (cwickert) [inline xml/deployment_installation.xml]: guimenu (and optionally replacable)

File: xml/deployment_installation.xml [hunk 12]
```diff
@@ -825,9 +825,253 @@
...
+            This will become the password for the root account on all nodes in the
```
Comment (cwickert) [inline xml/deployment_installation.xml]: use the &rootuser; entity

File: xml/deployment_installation.xml [hunk 13]
```diff
@@ -825,9 +825,253 @@
...
+            stack.
+           </para>
+         <screen>
+&prompt.user;<command>cat caasp-environment.yaml</command>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: Not sure if we need the command here, just the file contents should suffice.

File: xml/deployment_installation.xml [hunk 14]
```diff
@@ -825,9 +825,253 @@
...
+---
+parameters:
+  root_password: <replaceable>linux</replaceable>
+  admin_flavor: m1.large
+...
```
Comment (cwickert) [inline xml/deployment_installation.xml]: Use `&hellip;` instead of three dots

File: xml/deployment_installation.xml [hunk 15]
```diff
@@ -825,9 +825,253 @@
...
+       <step>
+        <para>
+        Click <guimenu>Launch</guimenu>.
+        </para>
+       </step>
+       
+       <step>
+        <para>
+         After the cluster has been started and the cluster overview shows 
+         <literal>Create Complete</literal> you need to find the external IP
```
Comment (cwickert) [inline xml/deployment_installation.xml]: Is this a gui element? then use guimenu.

File: xml/deployment_installation.xml [hunk 16]
```diff
@@ -825,9 +825,253 @@
...
+       <title>Using the &ostack; CLI</title>
+       <note>
+        <para>
+         You need to have access to the <command>openstack</command> command-line 
+         tools. You can either access those via ssh on your &soc; admin server
+         or <link xlink:href="https://docs.openstack.org/newton/user-guide/common/cli-install-openstack-command-line-clients.html">
+         install a local <command>openstack</command> client</link>.
+        </para>
+        <para>
+         To use the local client you need to access <guimenu>Project -> Compute ->
```
Comment (cwickert) [inline xml/deployment_installation.xml]: proper arrows…

File: xml/deployment_installation.xml (hunk 8 above)
Comment (r0ckarong) [reply xml/deployment_installation.xml]: Good catch

File: xml/deployment_installation.xml (hunk 13 above)
Comment (r0ckarong) [reply xml/deployment_installation.xml]: I would have wanted a caption for the screen field here describing what file the outputs are from, that's why I used that command to make it clear.

**Comment by r0ckarong** [comment]
@cwickert @cjschroder All requested changes added, please check again.

File: xml/deployment_installation.xml [hunk 17]
```diff
@@ -825,14 +825,350 @@
...
+            The internal network range to be used inside the cluster
+           </para>
+          </listitem>
+         </varlistentry>
+         <varlistentry>
+          <term><literal>dns_nameserver</literal>
+          </term>
+          <listitem>
+           <para>
+            Internal nameserver for the cluster
```
Comment (cwickert) [inline xml/deployment_installation.xml]: 'name server', not 'nameserver'. Refer to the [terminology in the Style Guide](http://docserv.nue.suse.com/documents/Style_Guide/docu_styleguide/html/app.terminology.html#sec.terminology)

File: xml/deployment_installation.xml (hunk 13 above)
Comment (cwickert) [reply xml/deployment_installation.xml]: Personally I think we don't need the screen here at all.

But if you want it, I suggest using a title for the file name. As screens cannot have titles, you can wrap everything in &lt;example&gt;:
```
<example>
 <title>Example of a caasp-environment.yaml file</title>
 <screen>
 ...
 </screen>
</example>
```

Downside: Examples are listed in the appendix and as this warning box occurs twice, there will be two example caasp-environment.yaml files.

File: xml/deployment_installation.xml [hunk 18]
```diff
@@ -825,14 +825,350 @@
...
+       <step>
+        <para>
+        Click <guimenu>Launch</guimenu>.
+        </para>
+       </step>
+       
+       <step>
+        <para>
+         After the cluster has been started and the cluster overview shows 
+         <literal>Create Complete</literal>, you need to find the external IP
```
Comment (cwickert) [inline xml/deployment_installation.xml]: guimenu instead of literal?

File: xml/deployment_installation.xml [hunk 19]
```diff
@@ -825,14 +825,350 @@
...
+           from the GitHub repository.
+          </para>
+           <para>
+            You must make sure to replace the value for 
+            <literal>root_password</literal> with a secure password.
+            This will become the password for the &rootuser; account on all 
+            nodes in the stack.
+           </para>
+         <screen>
+---
```
Comment (cwickert) [inline xml/deployment_installation.xml]: remove this line?

File: xml/deployment_installation.xml [hunk 20]
```diff
@@ -825,14 +825,350 @@
...
+parameters:
+root_password: <replaceable>linux</replaceable>
+admin_flavor: m1.large
+&hellip;
+         </screen>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: Personally I think we don't need the screen here at all.

But if you want it, I suggest using a title for the file name. As screens cannot have titles, you can wrap everything in &lt;example&gt;:
```
<example>
 <title>Example of a caasp-environment.yaml file</title>
 <screen>
 ...
 </screen>
</example>
```

Downside: Examples are listed in the appendix and as this warning box occurs twice, there will be two example caasp-environment.yaml files.

File: xml/deployment_installation.xml [hunk 21]
```diff
@@ -825,14 +825,350 @@
...
+          <term><literal>Stack Name</literal>
+          </term>
+          <listitem>
+           <para>
+            Give your stack a name
+           </para>
+          </listitem>
+         </varlistentry>
+         <varlistentry>
+          <term><literal>Password</literal>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: line break

File: xml/deployment_installation.xml [hunk 22]
```diff
@@ -825,14 +825,350 @@
...
+       </para>
+      </step>
+
+       <step>
+        <para>
+         Now you need to define more information about the stack.
+        </para>
+        <variablelist>
+         <varlistentry>
+          <term><literal>Stack Name</literal>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: line break after &lt;term&gt;

File: xml/deployment_installation.xml [hunk 23]
```diff
@@ -825,14 +825,350 @@
...
+          <term><literal>Password</literal>
+          </term>
+          <listitem>
+           <para>
+            Your &soc; password
+           </para>
+          </listitem>
+         </varlistentry>
+         <varlistentry>
+          <term><literal>Image</literal>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: line break

File: xml/deployment_installation.xml [hunk 24]
```diff
@@ -825,14 +825,350 @@
...
+          </term>
+          <listitem>
+           <para>
+            Select the image your machines will be created from
+           </para>
+          </listitem>
+         </varlistentry>
+         <varlistentry>
+          <term><literal>root_password</literal>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: line break

File: xml/deployment_installation.xml [hunk 25]
```diff
@@ -825,14 +825,350 @@
...
+          </term>
+          <listitem>
+           <para>
+            Set the root password for your cluster machines
+           </para>
+          </listitem>
+         </varlistentry>
+         <varlistentry>
+          <term><literal>admin</literal>/master/worker_flavor</term>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: line break

File: xml/deployment_installation.xml [hunk 26]
```diff
@@ -825,14 +825,350 @@
...
+          <listitem>
+           <para>
+            Select the machine flavor for your nodes
+           </para>
+          </listitem>
+         </varlistentry>
+         <varlistentry>
+          <term><literal>worker_count</literal>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: line break

File: xml/deployment_installation.xml [hunk 27]
```diff
@@ -825,14 +825,350 @@
...
+          </term>
+          <listitem>
+           <para>
+            Number of worker nodes to be launched
+           </para>
+          </listitem>
+         </varlistentry>
+         <varlistentry>
+          <term><literal>external_net</literal>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: line break

File: xml/deployment_installation.xml [hunk 28]
```diff
@@ -825,14 +825,350 @@
...
+          </term>
+          <listitem>
+           <para>
+            Select an external network that your cluster will be reachable from
+           </para>
+          </listitem>
+         </varlistentry>
+         <varlistentry>
+          <term><literal>internal_net_cidr</literal>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: line break

File: xml/deployment_installation.xml [hunk 29]
```diff
@@ -825,14 +825,350 @@
...
+          </term>
+          <listitem>
+           <para>
+            The internal network range to be used inside the cluster
+           </para>
+          </listitem>
+         </varlistentry>
+         <varlistentry>
+          <term><literal>dns_nameserver</literal>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: line break

File: xml/deployment_installation.xml [hunk 30]
```diff
@@ -825,14 +825,350 @@
...
+       </step>
+       
+      </procedure>
+    </sect3>
+      
+      <sect3>
+       <title>Using the &ostack; CLI</title>
+       <note>
+        <para>
+         You need to have access to the &ostack; command-line
```
Comment (cwickert) [inline xml/deployment_installation.xml]: 80 chars?

File: xml/deployment_installation.xml [hunk 31]
```diff
@@ -825,14 +825,350 @@
...
+          </screen>
+         </step>
+
+        <step>
+         <para>
+          Find out which (external) IP address was assigned to the admin node
+          of your &productname; cluster (here: <literal>10.81.1.51</literal>).
+         </para>
+          <screen>
+&prompt.user;<command>openstack server list --name "admin"</command>
```
Comment (cwickert) [inline xml/deployment_installation.xml]: This screen is way to wide. Either run it again on a 80*25 terminal (or 100x25 if it doesn't fit) or manually shorten it here.

File: xml/deployment_installation.xml (hunk 21 above)
Comment (r0ckarong) [reply xml/deployment_installation.xml]: This is how xmlformat spat it out, same as following

File: xml/deployment_installation.xml (hunk 30 above)
Comment (r0ckarong) [reply xml/deployment_installation.xml]: The link itself is 99 chars long, what can I do?

File: xml/deployment_installation.xml (hunk 31 above)
Comment (r0ckarong) [reply xml/deployment_installation.xml]: What would "manually shorten" mean? I already removed all unnecessary whitespace from the output that didn't cull any information

Comment (r0ckarong) [reply xml/deployment_installation.xml]: The output on a narrower terminal
```
| 7969fa8c-6b55-419b-9586-90c612b5c95a | caasp3-stack-admin                     | ACTIVE | caasp3-stack-caasp-int-net=172.24.0.15, 10.81.1.51       | CaaSP-3                           | m1.large  |
```

**Comment by cwickert** [comment]
@cjschroder Please look over this once again and approve it if your concerns were addressed. I will then squash and merge. Thanks!

File: xml/deployment_installation.xml [hunk 32]
```diff
@@ -823,14 +823,318 @@
...
+          of your &productname; cluster (here: <literal>10.81.1.51</literal>).
+         </para>
+          <screen>
+&prompt.user;<command>openstack server list --name "admin" | awk 'FNR > 3 {print $4 $5 $9}'</command>
+caasp3-stack-admin|10.81.1.51
+         </screen>
+        </step>
+        <step>
+         <para>
+          Visit the external IP in your browser. You should see the &dashboard;
```
Comment (cjschroder) [inline xml/deployment_installation.xml]: "IP address" please :) "IP" and "IP address" are not the same.

Comment (r0ckarong) [reply xml/deployment_installation.xml]: done

**Pull request merged**